system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -241,10 +241,7 @@ export class TopSitePlaceholder extends React.PureComponent {
     this.onEditButtonClick = this.onEditButtonClick.bind(this);
   }
   onEditButtonClick() {
-    this.props.dispatch({
-      type: at.TOP_SITES_EDIT,
-      data: {index: this.props.index}
-    });
+    this.props.onEdit(this.props.index);
```

Comment:
Probably just make it conditionally call `this.props.onEdit` or do the existing `dispatch` ?

Reply:
@rlr ^^ ?

Reply:
yeah, right now clicking on the edit icon from outside the modal doesn't work. I wonder if there is a way we could just kill the whole edit modal concept. I'll discuss with @bryanbell next week.

Reply:
@Mardak @rlr: I'm confused.  If I'm understanding what you guys have said correctly, something should be busted with the patch as written.  In my local tree, clicking the hovering edit icon brings up a functional edit modal both when done in the normal topsites or in the edit mode modal.   Am I missing something?

Reply:
The hovering edit icon over the giant image for a given site, that is.

Reply:
This is what I see (and in the console I see: `JavaScript error: resource://activity-stream/data/content/activity-stream.bundle.js, line 2887: TypeError: this.props.onEdit is not a function`):

<img src="https://d2ffutrenqvap3.cloudfront.net/items/2y31002G2c2E393O1W0O/Screen%20Recording%202018-01-18%20at%2011.22%20AM.gif" />

Reply:
@rlr thanks!   I was clicking the wrong thing.